rbd/diff_iterate_by_id.go

```diff
@@ -0,0 +1,93 @@
+//go:build ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#undef _GNU_SOURCE
+#include <errno.h>
+#include <stdlib.h>
+#include <rbd/librbd.h>
+
+// inline wrapper to cast uintptr_t to void*
+static inline int wrap_rbd_diff_iterate3(rbd_image_t image,
+	uint64_t from_snap_id, uint64_t ofs, uint64_t len, uint8_t include_parent,
+	uint8_t whole_object, uintptr_t arg) {
+		return rbd_diff_iterate3(image, from_snap_id, ofs, len, include_parent,
```

Comment:
Linking against librdb will fail when `rbd_diff_iterate3` does not exists. The solution is to use the `dlsym` package and discover the `rbd_diff_iterate3` function dynamically.

You can see how that is done for [`rbd_clobe4`](https://github.com/ceph/go-ceph/blob/master/rbd/clone_image_by_id.go). There is some indirection involved, but it is possible to build go-ceph applications with the new `CloneImageByID` function, even if it is not part of librbd.